component/src/org/wso2/ballerina/connectors/oauth2/ClientConnector.bal

```diff
@@ -128,7 +128,7 @@ public connector ClientConnector (string baseUrl, string accessToken, string cli
 }
 
 function populateAuthHeader (http:Request request, string accessToken) {
-    if (accessTokenValue == null) {
+    if (accessTokenValue == "") {
```

Comment:
What's the reason behind changing the default value to empty string and comparing it against the empty string? The default value of strings is null. So unless the value is changed from somewhere else, there's virtually no difference between the 2 comparisons.

Reply:
According to the spec (https://github.com/ballerinalang/ballerina/blob/master/docs/lang-ref/types-variables.md) default value for string is empty string (not null). This if condition is not taken in any case since null is not equal to empty string.

Reply:
The spec at that link is actually outdated. The default value of strings was changed to `null` recently. The most up-to-date source regarding language syntax, constructs etc. is [Ballerina by Example](https://ballerinalang.org/docs/by-example/).

Reply:
Ack. Sorry I was using 0.95.6